ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -435,188 +643,338 @@ string get_login(string username, string password, string cap)
   return gs_str_submitted;
 }
 
-string get_string(string message,string def,string cap)
+string get_string(string str, string def)
 {
-  gs_cap = cap; gs_message = message; gs_def = def;
-  DialogBox(enigma::hInstance,"getstringdialog",enigma::hWnd,GetStrProc);
+  tstring tstrStr = widen(str);
```

Comment:
These lines appear exactly in the next four functions, at very least. They should all be calling `get_string` and then operating on the result.

Reply:
TFD uses roughly the same amount of code for these several functions. Did you want me fix that as well?